saw-core/src/Verifier/SAW/SharedTerm.hs

```diff
@@ -1077,62 +1084,88 @@ scDefTerm sc d = scGlobalDef sc (defIdent d)
 scApplyCtor :: SharedContext -> Ctor -> [Term] -> IO Term
 scApplyCtor sc c args = scCtorApp sc (ctorName c) args
 
+-- | Create a term from a sort.
 scSort :: SharedContext -> Sort -> IO Term
 scSort sc s = scFlatTermF sc (Sort s)
 
+-- | Create a literal term from a natural number.
 scNat :: SharedContext -> Natural -> IO Term
 scNat sc n = scFlatTermF sc (NatLit n)
 
+-- | Create a literal term from a string.
 scString :: SharedContext -> String -> IO Term
 scString sc s = scFlatTermF sc (StringLit s)
 
+-- | Create a term representing the global type of strings.
 scStringType :: SharedContext -> IO Term
 scStringType sc = scFlatTermF sc preludeStringType
 
+-- | Create a vector term from a type (as a term) and a list of terms.
 scVector :: SharedContext -> Term -> [Term] -> IO Term
 scVector sc e xs = scFlatTermF sc (ArrayValue e (V.fromList xs))
 
+-- | Create a record term from a map from field names to terms.
 scRecord :: SharedContext -> Map FieldName Term -> IO Term
 scRecord sc m = scFlatTermF sc (RecordValue $ Map.assocs m)
 
+-- | Create a record field access term from a term representing a record and
+-- a field name.
 scRecordSelect :: SharedContext -> Term -> FieldName -> IO Term
 scRecordSelect sc t fname = scFlatTermF sc (RecordProj t fname)
 
+-- | Create a term representing the type of a record from a list associating
+-- field names (as @String@s) and types (as @Term@s).
```

Comment:
It'd be nice to mention that the ordering of fields doesn't matter for record types (see the docstring for the `RecordType` constructor of the `FlatTermF` datatype)

Reply:
Also, you might consider using `'String'` and `'Term'` instead of `@String@` and `@Term@`, so that haddock will add hyperlinks (probably a good idea to do this consistently throughout). Note that you can also put single quotes inside `@` to hyperlink individual identifiers, e.g. `@'scRecordSelect' sc m@.